frontend/src/components/node/Details.tsx

```diff
@@ -32,37 +59,143 @@ export default function NodeDetails() {
+
+    cloneNode.spec.unschedulable = !cordon;
+    dispatch(
+      clusterAction(
+        () =>
+          apply(cloneNode.jsonData)
+            .then(() => {
+              setNode(cloneNode);
+            })
+            .finally(() => {
+              setisUpdatingNodeScheduleProperty(false);
+            }),
+        {
+          startMessage: cordon
+            ? t('Uncordoning node {{name}}...', {
+                name: node.metadata.name,
+              })
+            : t('Cordoning node {{name}}…', { name: node.metadata.name }),
+          successMessage: cordon
+            ? t('Uncordoned node {{name}}.', { name: node.metadata.name })
+            : t('Cordoned node {{name}}.', { name: node.metadata.name }),
+          errorMessage: cordon
+            ? t('Failed to uncordon node {{name}}.', { name: node.metadata.name })
+            : t('Failed to cordon node {{name}}.', { name: node.metadata.name }),
+        }
+      )
+    );
+  }
+
+  function getDrainNodeStatus(cluster: string, nodeName: string) {
+    setTimeout(() => {
+      drainNodeStatus(cluster, nodeName)
+        .then(data => {
+          if (data !== 'success') {
+            getDrainNodeStatus(cluster, nodeName);
+            return;
+          }
+          const cloneNode = _.cloneDeep(node);
+
+          cloneNode.spec.unschedulable = !node.spec.unschedulable;
+          setNode(cloneNode);
+        })
+        .catch(error => {
+          enqueueSnackbar(error.message, { variant: 'error' });
+        });
+    }, 1000);
+  }
+
+  function handleNodeDrain(node: Node) {
+    const cluster = getCluster();
+    if (!cluster) return;
+
+    setisNodeDrainInProgress(true);
+    dispatch(
+      clusterAction(
+        () =>
+          drainNode(cluster, node.metadata.name)
+            .then(() => {
+              getDrainNodeStatus(cluster, node.metadata.name);
+            })
+            .finally(() => {
+              setisNodeDrainInProgress(false);
+            }),
+        {
+          startMessage: t('Draining node {{name}}…', { name: node.metadata.name }),
+          successMessage: t('Drained node {{name}}.', { name: node.metadata.name }),
+          errorMessage: t('Failed to drain node {{name}}.', { name: node.metadata.name }),
+        }
+      )
+    );
+  }
+
   return (
-    <DetailsGrid
-      headerSection={item => (
-        <ChartsSection node={item} metrics={nodeMetrics} noMetrics={noMetrics} />
+    <>
+      <MainInfoSection
+        error={nodeError}
+        headerSection={item => (
+          <ChartsSection node={item} metrics={nodeMetrics} noMetrics={noMetrics} />
+        )}
+        resource={node}
+        actions={item => {
+          const cordon = item?.jsonData?.spec?.unschedulable;
+          const cordonOrUncordon = cordon ? t('Uncordon') : t('Cordon');
+          return [
+            <AuthVisible item={item} authVerb="update">
+              <Tooltip title={cordonOrUncordon as string}>
+                <IconButton
```

Comment:
I think this can be accomplished with the ActionButton just like you do for the Drain.